core/math/a_star_grid_2d.cpp

```diff
@@ -558,6 +558,24 @@ Vector2 AStarGrid2D::get_point_position(const Vector2i &p_id) const {
 	return _get_point_unchecked(p_id)->pos;
 }
 
+Vector<Vector2> AStarGrid2D::get_point_positions_in_region(const Rect2i &p_region) const {
+	ERR_FAIL_COND_V_MSG(dirty, Vector<Vector2>(), "Grid is not initialized. Call the update method.");
+	const Rect2i inter_region = region.intersection(p_region);
+
+	const int32_t end_x = inter_region.get_end().x;
+	const int32_t end_y = inter_region.get_end().y;
+
+	Vector<Vector2> positions;
+
+	for (int32_t y = inter_region.position.y; y < end_y; y++) {
+		for (int32_t x = inter_region.position.x; x < end_x; x++) {
+			positions.push_back(_get_point_unchecked(Vector2i(x, y))->pos);
```

Comment:
```suggestion
	const int32_t end_x = inter_region.get_end().x - region.position.x;
	const int32_t end_y = inter_region.get_end().y - region.position.y;

	Vector<Vector2> positions;
	positions.resize(inter_region.size.x * inter_region.size.y);
	Vector2 *ptrw = positions.ptrw();

	for (int32_t y = inter_region.position.y - region.position.y; y < end_y; y++) {
		for (int32_t x = inter_region.position.x - region.position.x; x < end_x; x++) {
			*ptrw++ = points[y][x].pos;
```
Try this perhaps, including avoiding COW stuff